R/mod_new_submissions.R

```diff
@@ -0,0 +1,143 @@
+# Module UI
+
+#' @title   mod_new_submissions_ui and mod_new_submissions_server
+#' @description  A shiny Module.
+#'
+#' @param id shiny id
+#' @param input internal
+#' @param output internal
+#' @param session internal
+#'
+#' @rdname mod_summary_snapshot
+#'
+#' @keywords internal
+#' @export 
+#' @importFrom shiny NS tagList 
+#' @import ggplot2
+#' @import plotly
+mod_new_submissions_ui <- function(id){
+  ns <- shiny::NS(id)
+  
+  shiny::tagList(
+    shinydashboard::dashboardPage(
+      shinydashboard::dashboardHeader(disable = T),
+      shinydashboard::dashboardSidebar(disable = T),
+      shinydashboard::dashboardBody(
+        shiny::fluidPage(
+          shinydashboard::box(
+            title = "Funding Partner",
+            width = 12,
+            solidHeader = T,
+            status = "primary",
+            shiny::textOutput(ns('funding_agency'))
+          ),
+          shinydashboard::box(
+            title = "New Files",
+            status = "primary",
+            solidHeader = TRUE,
+            width = 12,
+            collapsible = FALSE,
+            shiny::numericInput(
+              ns("new_files_day_choice"),
+              "Display files uploaded within the last N days:",
+              62,
+              min = 1,
+              step = 1
+            ),
+            DT::dataTableOutput(ns('new_files_dt')),
+          )
+        )
+      )
+    )
+  )
+}
+
+# Module Server
+
+#' @rdname mod_new_submissions
+#' @export
+#' @keywords internal
+
+mod_new_submissions_server <- function(
+  input, output, session, group_object, data_config
+){
+  ns <- session$ns
+  
+  output$funding_agency <- shiny::renderText({
+    print(glue::glue(
+      "You are now viewing recent additions to studies moderated by {group_object()$selected_group}"
+    ))
+  })
+  
+  new_files_table <- shiny::reactive({
+    
+    shiny::req(group_object(), data_config, input$new_files_day_choice)
+    
+    param_list <- purrr::pluck(
+      data_config,
+      "modules",
+      "new_submissions",
+      "outputs",
+      "new_files_table"
+    )
+    
+    minimum_date <- 
+      lubridate::now() - 
+      lubridate::ddays(input$new_files_day_choice)
+    
+    files_table <- group_object() %>%
+      purrr::pluck("files_table") %>% 
+      dplyr::filter(!!rlang::sym(param_list$filter_column) > minimum_date) %>% 
+      dplyr::arrange(!!rlang::sym(param_list$filter_column))
+    
+    studies_table <- group_object() %>%
+      purrr::pluck("studies_table") 
+
+    data1 <- files_table %>%
```

Comment:
Study Id "syn11374337" has recent files added to it (date- 8/27/2020) but are not showing up in the table of recent additions.

Reply:
@jaybee84 Could you give me an example of a file that is missing?

Reply:
@andrewelamb sorry I missed this message earlier. Here are some files that [were added on 7/31/2020](https://www.synapse.org/#!Synapse:syn11601451/tables/query/eyJzcWwiOiJTRUxFQ1QgKiBGUk9NIHN5bjExNjAxNDUxIiwgInNlbGVjdGVkRmFjZXRzIjpbeyJjb25jcmV0ZVR5cGUiOiJvcmcuc2FnZWJpb25ldHdvcmtzLnJlcG8ubW9kZWwudGFibGUuRmFjZXRDb2x1bW5WYWx1ZXNSZXF1ZXN0IiwgImNvbHVtbk5hbWUiOiJwYXJlbnRJZCIsICJmYWNldFZhbHVlcyI6WyJzeW4yMjI4MTA2MiJdfV0sICJpbmNsdWRlRW50aXR5RXRhZyI6dHJ1ZSwgIm9mZnNldCI6MCwgImxpbWl0IjoyNX0=)

Reply:
These files don't have a fundinAgency. In fact there are 1767 files total without a fundingAgency

Reply:
Ah right! The new files are not annotated and so are missing the funding Agency annotation. 

While we can potentially use the projectIds to pull unannotated files, after discussing with @allaway a better solution seems to be to add fundingAgency annotation directly to the files_table.rds file which will be updated regularly through the files.R script.

I am putting together a short script to add to files.R and will push to this branch soon :)

Reply:
I might be confused! The files synapse table already has a fundingAgency column.

Reply:
Just to clarify:

> @allaway a better solution seems to be to add fundingAgency annotation directly to the files_table.rds file which will be updated regularly through the files.R script.
>
>I am putting together a short script to add to files.R and will push to this branch soon :)


My thought was that we put together a script to just actually annotate these files with the correct `studyId` `studyName` and `fundingAgency` on synapse based on the parentId and/or projectId- (i.e. upstream of generating `files_table.rds`, not as part of generating that file)

Reply:
@allaway I agree, but wouldn't thing to fix be the files synapse table(syn22282675) since it's the source of data for the files .rds file?

Reply:
Another possibly dumb suggestion/question. Why does the files table have a fundingAgency column? Shouldn't a file inherit it's funding agency from the study belongs to. ie. if file "A" belongs to study "B", and study "B" has NTAP as it's funding agency, that wopuld imply file "A" also has NTAP as its funding agency. Having a files funding agency in two different places allows things like this to happen where it has two sources of truth that can disagree with each other.

Reply:
> @allaway I agree, but wouldn't thing to fix be the files synapse table(syn22282675) since it's the source of data for the files .rds file?

I'm guessing this table is just an RDS version of our portal fileview, which is the core source of ground truth for NF files (https://www.synapse.org/#!Synapse:syn16858331/tables/), so fixing the file annotations and updating syn22282675 from syn16858331 would be the same thing. I think we're on the same page :)

>Another possibly dumb suggestion/question. Why does the files table have a fundingAgency column? Shouldn't a file inherit it's funding agency from the study belongs to. ie. if file "A" belongs to study "B", and study "B" has NTAP as it's funding agency, that wopuld imply file "A" also has NTAP as its funding agency. Having a files funding agency in two different places allows things like this to happen where it has two sources of truth that can disagree with each other.

Not a dumb question. You are right that funding agency is probably in all cases a study property and not a file property. But, we want Portal users to be able to filter files by funder, and we can't do joins in synapse, so we have to duplicate information. This same problem exists for Individuals/Patients, cell lines, etc. 

As I understand it there is a longer term plan to be able to define these sorts of objects in synapse so that the, but that's not a thing yet. :)

Reply:
@allaway : yes that will definitely be in the purview of annotation of the files themselves. I suggested a downstream version here so that we don't accidentally overwrite the original portal-files table while testing out projectLive.. :)
Once tested in the purview of projectLive, the same script can then be reused for annotation down the road.

Reply:
@andrewlamb : 1. While fixing the fundingAgency annotation at the file level or portal-files fileview level would be ideal, I am trying not to change anything in the portal-files fileview (the source of the rds file) since any mishaps there would directly affect the data portal.

2. Your suggestion is exactly what Robert and I discussed earlier today, and I was thinking of adding as a Jira issue for platform :) seems like there should be a way to add/inherit default values for certain columns. But that is not the case today.

Reply:
@allaway I think we are saying the same thing as well :). 

Given the portal based restrictions it seems like the files synapse table shouldn't be hand-annotated(at least for funding agency) but should be given the same funding agency as their study via some script to ensure data integrity. I don't know enough about portals and their tables to say much more about that though.

Reply:
Seems like we were typing in responses at the same time. 

>@allaway I agree, but wouldn't thing to fix be the files synapse table(syn22282675) since it's the source of data for the files .rds file?

>I'm guessing this table is just an RDS version of our portal fileview, which is the core source of ground truth for NF files (https://www.synapse.org/#!Synapse:syn16858331/tables/), so fixing the file annotations and updating syn22282675 from syn16858331 would be the same thing. I think we're on the same page :)

>Another possibly dumb suggestion/question. Why does the files table have a fundingAgency column? Shouldn't a file inherit it's funding agency from the study belongs to. ie. if file "A" belongs to study "B", and study "B" has NTAP as it's funding agency, that wopuld imply file "A" also has NTAP as its funding agency. Having a files funding agency in two different places allows things like this to happen where it has two sources of truth that can disagree with each other.

>Not a dumb question. You are right that funding agency is probably in all cases a study property and not a file property. But, we want Portal users to be able to filter files by funder (and to be able to list files for a specific funder, e.g. here), and we can't do joins in synapse, so we have to duplicate information. This same problem exists for Individuals/Patients, cell lines, etc.

>As I understand it there is a longer term plan to be able to define these sorts of objects in synapse so that we could define these properties as you suggest above, but that's not a thing yet. :)

Reply:
Just to clarify the above discussion and make sure we are all thinking of the same things:

For projectLive to act as an internal tracker for new and unannotated files submitted to NF projects without proper notification: 
1. All files within any NF project need to be pulled into the app according to fundingAgency annotation
2. The newest commit adds missing fundingAgency annotation (matched by studyId/projectId) to all files independent of the annotations in the portal fileview (source of the files.RDS file).
3. Unannotated new files should surface in the new_submissions module with "Not Annotated" values in the unannotated fields (this would help the data curator/tracker to find the right files, go in and add required annotations at the fileview level)

For annotation of the files:
Separate annotation scripts will be available to the data curator to add the proper annotations in the vacant fields. This will be done separately at the individual project fileview level or portal fileview level

Reply:
@jaybee84 @allaway 

Once again I may not be grasping something about portals. In my opinion if you need to have fundingAgency in every synapse table, in cases where they are inconsistent(like the files that that are unannotated), wouldn't the solution be to fix those in the synapse table? Presumably if you are using them in the portal, they should be correct there right? Then when we create the rds files from the sunapse table it would be correct also.